From 51576705fa505ea50c2675447119365c43c28bfc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:10 -0400
Subject: [usb] xhci: fix SCT_FOR_CTX(p) macro

Message-id: <1410887463-20674-179-git-send-email-dzickus@redhat.com>
Patchwork-id: 95310
O-Subject: [RHEL7 PATCH 178/591] xhci: fix SCT_FOR_CTX(p) macro
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 63a67a72d63dd077c2313cf19eb29d8e4bfa6963
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Aug 26 23:29:47 2013 +0300

    xhci: fix SCT_FOR_CTX(p) macro

    SCT_FOR_CTX(p) is defined as (((p) << 1) & 0x7) in which case if we want
    to set the stream context type to SCT_SSA_256 i.e 0x7 (although secondary
    stream arrays are not yet supported) using this macro definition we will
    get actually 0x6 which is not what we want.

    This patch fixes the above issue by defining the SCT_FOR_CTX(p) macro as
    (((p) & 0x7) << 1)

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.h b/drivers/usb/host/xhci.h
index 03c74b7..a33e8b5 100644
--- a/drivers/usb/host/xhci.h
+++ b/drivers/usb/host/xhci.h
@@ -752,7 +752,7 @@ struct xhci_stream_ctx {
 };
 
 /* Stream Context Types (section 6.4.1) - bits 3:1 of stream ctx deq ptr */
-#define SCT_FOR_CTX(p)  (((p) << 1) & 0x7)
+#define SCT_FOR_CTX(p)  (((p) & 0x7) << 1)
 /* Secondary stream array type, dequeue pointer is to a transfer ring */
 #define SCT_SEC_TR  0
 /* Primary stream array type, dequeue pointer is to a transfer ring */
-- 
1.7.1